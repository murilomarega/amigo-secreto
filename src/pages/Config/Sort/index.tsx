import React, { FC, useState } from "react";
import { useDrawResul } from "../../../state/hooks/useDrawResult";
import { useParticipantsList } from "../../../state/hooks/useParticipantsList";
import "./styles.css";

const Sort: FC = () => {
  const players = useParticipantsList();
  const [player, setPlayer] = useState<string>("");
  const [secretFriend, setSecretFriend] = useState<string>("");

  const result = useDrawResul();

  const draw = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (result.has(player)) {
      setSecretFriend(result.get(player)!);
    }
  };

  return (
    <section className="sorteio">
      <h2>Quem vai tirar o papelzinho?</h2>
      <form onSubmit={draw}>
        <select
          required
          name="players"
          id="players"
          placeholder="Selecione o seu nome"
          value={player}
          onChange={(event) => setPlayer(event.target.value)}
        >
          <option>Selecione seu nome</option>
          {players.map((name) => (
            <option key={name} value={name}>
              {name}
            </option>
          ))}
        </select>
        <p>Clique em em sortear para ver quem é seu amigo secreto!</p>
        <button className="botao-sortear">Sortear</button>
        {secretFriend && <p role="alert">{secretFriend}</p>}
      </form>
      <footer className="sorteio">
        <img
          src="/images/aviao.png"
          className="aviao"
          alt="Um desenho de um avião de papel"
        />
      </footer>
    </section>
  );
};

export default Sort;
