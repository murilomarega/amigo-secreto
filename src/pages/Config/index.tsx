import React, { FC } from "react";
import Card from "../../components/Card";
import Footer from "../../components/Footer";
import Form from "../../components/Form";
import ParticipantsList from "../../components/ParticipantsList";

const Config: FC = () => {
  return (
    <Card>
      <section>
        <h2>Vamos come√ßar!</h2>
        <Form />
        <ParticipantsList />
        <Footer />
      </section>
    </Card>
  );
};

export default Config;
