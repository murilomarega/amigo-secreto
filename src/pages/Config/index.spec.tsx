import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import Config from ".";

const mockNavigation = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigation,
  };
});

describe("Config page", () => {
  test("page must be rendered correctly", () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
