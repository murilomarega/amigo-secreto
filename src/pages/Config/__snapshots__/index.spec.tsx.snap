// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config page page must be rendered correctly 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Vamos come√ßar!
      </h2>
      <form>
        <div
          class="grupo-input-btn"
        >
          <input
            placeholder="Insira o nome dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="rodape-configuracoes"
      >
        <button
          class="botao"
          disabled=""
        >
          Iniciar sorteio
        </button>
        <img
          alt="Sacolas de compras"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
